addons/mail/tests/test_mail_template.py

```diff
@@ -435,3 +435,75 @@ def test_switch_icp_value(self):
+            {'name': f'Custom Template {i + 1}', 'description': f'Desc {i + 1}'}
+            for i in range(5)
+        ])
+
+        # 4 Base templates with XML ID
+        self.base_templates = self.mail_template.create([
+            {'name': f'Base Template {i + 1}', 'description': f'Desc Base {i + 1}'}
+            for i in range(4)
+        ])
+
+        for template in self.base_templates:
+            self.model_data.create({
+                'name': f'mail_template_{template.id}',
+                'module': 'test_module',
+                'model': 'mail.template',
+                'res_id': template.id
+            })
+
+    def test_search_template_category(self):
+        # Search by hidden templates
+        hidden_domain = self.mail_template._search_template_category('in', ['hidden_template'])
+        hidden_templates = self.mail_template.search(hidden_domain) - self.existing
+        self.assertEqual(len(hidden_templates), len(self.hidden_templates), "Hidden templates count mismatch")
+
+        # Search by base templates
+        base_domain = self.mail_template._search_template_category('in', ['base_template'])
+        base_templates = self.mail_template.search(base_domain) - self.existing
+        self.assertEqual(len(base_templates), len(self.base_templates), "Base templates count mismatch")
+
+        # Search by custom templates
+        custom_domain = self.mail_template._search_template_category('in', ['custom_template'])
+        custom_templates = self.mail_template.search(custom_domain) - self.existing
+        self.assertEqual(len(custom_templates), len(self.custom_templates), "Custom templates count mismatch")
+
+        # Combined search
+        combined_domain = self.mail_template._search_template_category('in', ['hidden_template', 'base_template', 'custom_template'])
+        combined_templates = self.mail_template.search(combined_domain) - self.existing
+        total_templates = len(self.hidden_templates) + len(self.base_templates) + len(self.custom_templates)
+        self.assertEqual(len(combined_templates), total_templates, "Combined templates count mismatch")
+
+        # Search with 'not in' operator
+        not_in_domain = self.mail_template._search_template_category('not in', ['hidden_template'])
+        not_in_templates = self.mail_template.search(not_in_domain) - self.existing
+        expected_templates = len(self.base_templates) + len(self.custom_templates)
+        self.assertEqual(len(not_in_templates), expected_templates, "Not in templates count mismatch")
+
+        # Search with 'not in' operator
+        not_in_domain = self.mail_template._search_template_category('not in', ['hidden_template', 'base_template'])
+        not_in_templates = self.mail_template.search(not_in_domain) - self.existing
+        expected_templates = len(self.custom_templates)
+        self.assertEqual(len(not_in_templates), expected_templates, "Not in multi templates count mismatch")
```

Comment:
For sanity, we could also check that "hidden_templates" etc.. effectively map to what we expect. This way we also test that the compute matches the search

Reply:
Yes was thinking the same this night.
And it permits to detect that a template without description was wrongly considered as hidden whatever if xmlid or not. (Maybe we should change this behavior for compute too, but not the purpose here)

So I have adapted at the same time the _search method.